### ISSUE TITLE
Again the freelancer rates

### ISSUE THREAD

**Issue by spapas** [exercism/elixir#863]
Hello, I was trying to solve the freelancer rates exercise. The solution  that passed all the tests is this one:

```
defmodule FreelancerRates do
  def daily_rate(hourly_rate) do
    8.0 * hourly_rate
  end
  def apply_discount(before_discount, discount) do
    before_discount * (1-discount/100)
  end
  def monthly_rate(hourly_rate, discount) do
    ceil(apply_discount(22 * daily_rate(hourly_rate), discount) ) 
  end
  def days_in_budget(budget, hourly_rate, discount) do
    Float.floor(budget / apply_discount(daily_rate(hourly_rate), discount), 1)
  end
end
```

This is not a valid solution because instead of round and Floor.round I had to use ceil and Float.floor to make tests pass. It seems that there's something fishy with the tests. Of course as it's already described in a different issue using floats for money is a major no-no and should not be taught to new programmers.

**Comment by angelikatyborska** [comment]
Hi! Thanks for raising the issue. Can you give me a longer explanation for this?

> This is not a valid solution because instead of round and Floor.round I had to use ceil and Float.floor to make tests pass.

The "ideal" solution (see https://github.com/exercism/elixir/blob/43dae088135a739b445c407681c2fdf6a92400d5/exercises/concept/freelancer-rates/.meta/exemplar.ex) uses `Kernel.trunc` + `Float.ceil` in the function `monthly_rate`. That doesn't sound much different than using just `Kernel.ceil`. Why do say that's not valid?

> there's something fishy with the tests

Would this change help?
https://github.com/exercism/elixir/pull/858

**Comment by spapas** [comment]
Hello, the problem is that the instructions don't keep up with the tests. The instructions for task 3 mention

> The returned monthly rate should be rounded up to the nearest integer.

and for task 4: 

> The returned number of days should be rounded down to one decimal place.

Rounding means using the round and Floor.round functions i.e when you have 3.4 you will get 3, for 3.5 you will get 4 and for 3.6 you will also get 4. The functions truct/ceil and floor have a different behavior that is *not* the same as rounding a number. 

So either the tests must be fixed so they will pass when users use round (https://hexdocs.pm/elixir/1.12/Kernel.html#round/1) for integers and Float.round (https://hexdocs.pm/elixir/1.12/Float.html#round/2) for floats *or* the instructions changed to describe the proper behavior (for example instead of "The returned monthly rate should be rounded up to the nearest integer." it could say "The returned monthly rate should be the smallest integer that is bigger than the result (for ceil)". 

The solution you mention may or may not help I can't test it right now... However even if it works like this, I still believe that you should stay clear from ceil/floor/trunc when you are writing "round" in the instructions.

**Comment by angelikatyborska** [comment]
I think we interpret the instructions differently. This the wording I have used in the instructions:

- round up == ceil
- round down == floor
- round == round

Wikipedia seems to agree that that's equivalent: https://en.wikipedia.org/wiki/Rounding#Rounding_down

I did that because I don't think that "ceiling" and "flooring" are verbs 🤔. For example this sounds totally wrong to me:

> The returned monthly rate should be ceiled to the nearest integer.

How else should that be phrased? Maybe like this, adding the alternative phrase in parenthesis? 

> The returned monthly rate should be rounded up (take the ceiling) to the nearest integer.

> The returned number of days should be rounded down (take the floor) to one decimal place.

**Comment by spapas** [comment]
I just saw that the instructions mention round up and down. I was just seeing the verb round without the directions. I am sorry that I missed it. English is not my primary language and I just skipped the up/down part.

The instructions are fine as they are but maybe for careless readers like my something like

> The returned monthly rate should be rounded up (take the ceiling) to the nearest integer.

or even just

> The returned monthly rate should be rounded up (ceil) to the nearest integer.

would definitely help :)

Sorry once again for misreading the instructions and raising this issue.
